emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +102,90 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    var raycaster = new THREE.Raycaster();
+    var mouse = new THREE.Vector2();
+
+    //initialize subscribers for event callbacks
+    this.EVENTS = ['click', 'dblclick'];
+    var subscribers = {};
+    for (var i = 0; i < this.EVENTS.length; i++) {
+      subscribers[this.EVENTS[i]] = [];
+    }
+
+    //create helper function for callbacks
+    var eventCallback = function(eventType, event) {
+      event.preventDefault();
+      //dont do anything if no subscribers
+      if (subscribers[eventType].length === 0) {
+        return;
+      }
+
+      var element = scope.renderer.domElement;
+      mouse.x = ((event.clientX - element.offsetLeft) / element.width) * 2 - 1;
+      mouse.y = -((event.clientY - element.offsetTop) / element.height) * 2 + 1;
+
+      raycaster.setFromCamera(mouse, scope.camera);
```

Comment:
From this, it seems like the selection will not catch a single or double click, as you need to get the entire objet in the selection space to select it. What we can do is for click events use raycasting, and for mouseDown/mouseUp use the marquee. this way we can merge this PR for click events, and then easily add on the marquee as a single new helper function for the two events. Sound good?